core/unix/signal.c

```diff
@@ -2434,6 +2449,10 @@ handle_sigprocmask(dcontext_t *dcontext, int how, kernel_sigset_t *app_set,
     thread_sig_info_t *info = (thread_sig_info_t *)dcontext->signal_field;
     int i;
     kernel_sigset_t safe_set;
+
+    /* We assume any change to the mask ends a handler: e.g., sigprocmask. */
+    info->in_app_handler = false;
```

Comment:
Technically yes but it seems a weird thing to do except on exit (e.g., via siglongjmp) since the handler setup mechanism installs a mask for the handler.  I would be quite surprised to see it in the middle.

Reply:
Added a comment.